b2luigi/batch/processes/gbasf2.py

```diff
@@ -0,0 +1,764 @@
+        self.retries_file_path = os.path.join(self.log_file_dir, "n_retries_by_grid_job.json")
+        if os.path.isfile(self.retries_file_path):
+            with open(self.retries_file_path, "r") as retries_file:
+                retries_from_file = json.load(retries_file)
+                self.n_retries_by_job.update(retries_from_file)
+
+        # Store dictionary with n_jobs_by_status in attribute to check if it changed,
+        # useful for printing job status on change only
+        self._n_jobs_by_status = ""
+
+        # Store whether the job had already been successful in a variable b/c
+        # there's actions we want to do only the first time that
+        # ``get_job_status`` returns a success.
+        self._project_had_been_successful = False
+
+    def get_job_status(self):
+        """
+        Get overall status of the gbasf2 project.
+
+        First obtain the status of all (sub-) jobs in a gbasf2 project, similar
+        to ``gb2_job_status``, and return an overall project status, e.g. when
+        all jobs are done, return ``JobStatus.successful`` to show that the
+        gbasf2 project succeeded.
+
+        The status of each individual job can be one of::
+
+            [Submitting, Submitted, Received, Checking, Staging, Waiting, Matched, Rescheduled,
+             Running, Stalled, Completing, Done, Completed, Failed, Deleted, Killed]
+
+        (Taken from  https://github.com/DIRACGrid/DIRAC/blob/rel-v7r1/WorkloadManagementSystem/Client/JobStatus.py)
+
+        """
+        # If project is does not exist on grid yet, so can't query for gbasf2 project status
+        if not check_project_exists(self.gbasf2_project_name, self.gbasf2_env, dirac_user=self.dirac_user):
+            raise RuntimeError(
+                f"Could not find any jobs for project {self.gbasf2_project_name} on the grid.\n" +
+                "Probably there was an error during the project submission when running the gbasf2 command.\n" +
+                "Try if you can run the gbasf2 command used manually in a terminal with gbasf2 set up:\n" +
+                " ".join(self._build_gbasf2_submit_command())
+            )
+
+        job_status_dict = get_gbasf2_project_job_status_dict(self.gbasf2_project_name, self.gbasf2_env, dirac_user=self.dirac_user)
+        n_jobs_by_status = Counter()
+        for _, job_info in job_status_dict.items():
+            n_jobs_by_status[job_info["Status"]] += 1
+
+        # print summary of jobs in project if setting is set and job status changed
+        if (get_setting("gbasf2_print_status_updates", default=True, task=self.task) and
+                n_jobs_by_status != self._n_jobs_by_status):
+            time_string = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+            job_status_string = str(dict(sorted(n_jobs_by_status.items()))).strip("{}")
+            print(f"Jobs in gbasf2 project \"{self.gbasf2_project_name}\" at {time_string}: {job_status_string}")
```

Comment:
Would you think this is useful for all batch processes, e.g. also for htcondor? Then we could implement this in the calling classes.